osdep/io.h

```diff
@@ -34,14 +34,25 @@
 
 #ifdef __ANDROID__
 #  include <unistd.h>
+#  include <stdio.h>
 #  ifdef lseek
 #   undef lseek
 #  endif
 #  define lseek(f,p,w) lseek64((f), (p), (w))
 #  ifdef fseeko
 #   undef fseeko
 #  endif
-#  define fseeko(f,p,w) lseek(fileno(f),(p),(w))
+
+static inline int mp_fseeko(FILE* fp, off64_t offset, int whence) {
+    int ret = -1;
+    if ((ret = fflush(fp)) != 0) {
+        return ret;
+    }
```

Comment:
Why is this needed?

Reply:
We're touching the file descriptor indecently from "underneath" the `FILE *` itself.

Discussion from last night with @rcombs:
```
01:36 < rcombs> JEEB: yeah that should work as long as nothing seeks,
                doesn't immediately seek back, and then attempts to read or
                write without first flushing
01:36 < rcombs> (i.e. as long as nothing produces a discrepancy between the
                FILE's buffer and the fd's position)
01:48 < rcombs> might want to fflush() to be sure
01:50 < rcombs> see what fseeko actually does:
https://github.com/android/platform_bionic/blob/38bcf2b704eb5038668759d2dd37115aea1dc437/libc/stdio/stdio.cpp#L496
01:51 < rcombs> (hint: it involves [basically] fflush())
01:52 < JEEB> so basically fflush and then seek?
01:52 < JEEB> or the other way+
01:53 < rcombs> flush then seek
01:53 < JEEB> k
01:56 < rcombs> also it won't work if anything uses ungetc, and doesn't
                clear the EOF flag
01:57 < rcombs> clearerr() will do that
01:57 < rcombs> (clear the EOF flag, that is)
```

Reply:
Makes total sense. I assume there's no FILE API for this. Go on then.